src/content/plugins/internal-plugins.md

```diff
@@ -0,0 +1,335 @@
+---
+title: Internal webpack plugins
+contributors:
+  - EugeneHlushko
+  - ooflorent
+  - Legends
+---
+
+This is a list of plugins which are used by webpack internally.
+
+W>You should only care about them if you are building your own compiler based on webpack, or introspect the internals.
+
+categories of internal plugins:
+
+* [environment](#environment)
+* [compiler](#compiler)
+* [entry](#entry)
+* [output](#output)
+* [source](#source)
+* [optimize](#optimize)
+
+## environment
+
+Plugins affecting the environment of the compiler.
+
+### NodeEnvironmentPlugin
+
+`node/NodeEnvironmentPlugin()`
+
+Applies node.js style filesystem to the compiler.
+
+## compiler
+
+Plugins affecting the compiler
+
+### CachePlugin
+
+`CachePlugin([cache])`
+
+Adds a cache to the compiler, where modules are cached.
+
+You can pass a `cache` object, where the modules are cached. Otherwise one is created per plugin instance.
+
+### ProgressPlugin
+
+`ProgressPlugin(handler)`
+
+Hook into the compiler to extract progress information. The `handler` must have the signature `function(percentage, message)`. It's called with `0 <= percentage <= 1`. `percentage == 0` indicates the start. `percentage == 1` indicates the end.
+
+### RecordIdsPlugin
+
+`RecordIdsPlugin()`
+
+Saves and restores module and chunk ids from records.
+
+## entry
+
+Plugins, which add entry chunks to the compilation.
+
+### SingleEntryPlugin
+
+`SingleEntryPlugin(context, request, chunkName)`
+
+Adds an entry chunk on compilation. The chunk is named `chunkName` and contains only one module (plus dependencies). The module is resolved from `request` in `context` (absolute path).
+
+### MultiEntryPlugin
+
+`MultiEntryPlugin(context, requests, chunkName)`
+
+Adds an entry chunk on compilation. The chunk is named `chunkName` and contains a module for each item in the `requests` array (plus dependencies). Each item in `requests` is resolved in `context` (absolute path).
+
+### PrefetchPlugin
+
+`PrefetchPlugin(context, request)`
+
+Prefetches `request` and dependencies to enable a more parallel compilation. It doesn't create any chunk. The module is resolved from `request` in `context` (absolute path).
+
+## output
+
+### FunctionModulePlugin
+
+`FunctionModulePlugin(context, options)`
+
+Each emitted module is wrapped in a function.
+
+`options` are the output options.
+
+If `options.pathinfo` is set, each module function is annotated with a comment containing the module identifier shortened to `context` (absolute path).
+
+### JsonpTemplatePlugin
+
+`JsonpTemplatePlugin(options)`
+
+Chunks are wrapped into JSONP-calls. A loading algorithm is included in entry chunks. It loads chunks by adding a `<script>` tag.
```

Comment:
Don't we have some docs regarding preload/prefetch? We are talking about the same right?

Reply:
yes but this page just enlists all internals

Reply:
Exactly, a link would be useful.